src/nvim/os/input.c

```diff
@@ -98,14 +98,20 @@ static void create_cursorhold_event(bool events_enabled)
 
 /// Low level input function
 ///
-/// wait until either the input buffer is non-empty or , if `events` is not NULL
+/// wait until either the input buffer is non-empty or, if `events` is not NULL
 /// until `events` is non-empty.
 int os_inchar(uint8_t *buf, int maxlen, int ms, int tb_change_cnt, MultiQueue *events)
 {
   if (maxlen && rbuffer_size(input_buffer)) {
     return (int)rbuffer_read(input_buffer, (char *)buf, (size_t)maxlen);
   }
 
+  // There is no need for CTRL-C to interrupt something, don't let
+  // it set got_int when it was mapped.
```

Comment:
what does this mean? If there's "no need" then we do we check if it's mapped? Why not always disable it? 

Is "low risk" a better way to word this?

```suggestion
  // No risk of a UI flood, so disable CTRL-C "interrupt" behavior if it's mapped.
```

Reply:
This comment comes from the code removed in <https://github.com/neovim/neovim/commit/541eaf598c25e0b853ef441b57c9f7d1bbf3450c>. There is also a `wtime == -1 || wtime > 100L` in the removed code, but I'm not sure if I should add that check or not, as adding this check will prevent `<C-C>` mapping from working if it is put into input buffer immediately after other keys.

Reply:
I _think_ my suggestion makes sense then, but I'll leave it up to you.